### ISSUE TITLE
The status of the container is stuck in stopping.

### ISSUE THREAD

**Issue by wqshr12345** [containers/podman#16641]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

I was connecting to one of my containers by ssh,and looking at some code above, then I'm off to eat :(. When I came back after lunch, I found that ssh was disconnected and container was not accessible.
I tried podman stop、podman restart and podman kill but nothing worked.
I guarantee that after I used ssh to connect to the container, I only looked at some code and did not run any programs. **My appeal is how to make the status of this container no longer stop, so that I can use it again.**
Thank you!
**Steps to reproduce the issue:**

1.Use ssh to connect to a container:
ssh username@IP -p container‘s port
2.go to lunch :(

**Describe the results you received:**
The status of this container is always stopping.
![75@)3~9NHGYFQGHTWYG0_NQ](https://user-images.githubusercontent.com/43242755/204189694-d9c96ef1-b611-4264-a0ab-9283738e2ecc.png)
xx represents my container id.
podman stop xx
Error: can only stop created or running containers. xx is in state stopping: container state improper

podman restart xx
Error: unable to restart a container in a paused or unknown state: container state improper

podman kill xx
ERRO[0000] container not running                        
Error: error sending signal to container xx : `/usr/bin/runc kill xx 9` failed: exit status 1
**Describe the results you expected:**
The status of the container is stopped or running

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
podman version 4.0.0-dev
```

**Output of `podman info`:**

```
host:
  arch: amd64
  buildahVersion: 1.23.0
  cgroupControllers:
  - cpuset
  - cpu
  - cpuacct
  - blkio
  - memory
  - devices
  - freezer
  - net_cls
  - perf_event
  - net_prio
  - hugetlb
  - pids
  - rdma
  cgroupManager: systemd
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.2-1.1.6.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.2, commit: '
  cpus: 80
  distribution:
    distribution: '"centos"'
    version: "8"
  eventLogger: journald
  hostname: localhost.localdomain
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 4.18.0-348.7.1.el8_5.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 56096075776
  memTotal: 134365310976
  ociRuntime:
    name: crun
    package: crun-1.0-1.module_el8.5.0+911+f19012f9.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.0
      commit: 139dc6971e2f1d931af520188763e984d6cdfbf8
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT,CAP_AUDIT_CONTROL,CAP_AUDIT_WRITE
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.8-1.module_el8.5.0+890+6b136101.x86_64
    version: |-
      slirp4netns version 1.1.8
      commit: d361001f495417b880f20329121e3aa431a8f90f
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.1
  swapFree: 694501376
  swapTotal: 4294963200
  uptime: 330h 10m 25.02s (Approximately 13.75 days)
plugins:
  log:
  - k8s-file
  - none
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
store:
  configFile: /etc/containers/storage.conf
  containerStore:
    number: 21
    paused: 0
    running: 4
    stopped: 17
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 23
  runRoot: /var/run/containers/storage
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.0.0-dev
  Built: 1633015040
  BuiltTime: Thu Sep 30 23:17:20 2021
  GitCommit: ""
  GoVersion: go1.16.7
  OsArch: linux/amd64
  Version: 4.0.0-dev
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman` or `brew info podman`):**

```
podman-4.0.0-0.10.module_el8.6.0+944+d413f95e.x86_64
```

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/main/troubleshooting.md)**

No

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by vrothberg** [comment]
Thanks for reaching out, @wqshr12345!

There were a number of fixes in that area since Podman 4.0.  If possible, I recommend upgrading to a newer version of Podman.  Version v4.3 should be in CentOS Stream by now.

In case you find an easier reproducer (I've only lunch per day :), please drop it here.  For now, I am going to close the issue as I believe it has been fixed.

**Comment by vrothberg** [comment]
@gbraad can you be more specific about the exact behavior?  If `conmon` gets killed, the container can remain in the "stopping" state but a `podman stop` or `rm/kill` should clean it up.

**Comment by gbraad** [comment]
Issuing a stop command didn't do anything. I had to rm -f the container. But that sounds like a bad workaround. A start is impossible as it is in the wrong state. Reboot was the only thing that recovered the situation without having to remove the container.

**Comment by vrothberg** [comment]
I'll prepare a fix after lunch :heavy_check_mark:

**Comment by vrothberg** [comment]
OK, I thought to have seen a wrong check but Podman allows for stopping a container in the "stopping" state to avoid such a situation.

@gbraad if you run into this situation again, please collect debug logs via `podman --log-level=debug`.  Access to the machine may help as well.

**Comment by mwri** [comment]
Hit this today. Have podman version 4.2.0, the server was rebooted, and afterwards running `podman ps` yielded:

> ERRO[0000] invalid internal status, try resetting the pause process with "podman system migrate": could not find any running process: no such process

Running `podman system migrate` fixed that, but now the containers are now all in "stopping" state, can't start them, can't stop them.

Force remove and recreation worked of course.

**Comment by lainwir3d** [comment]
I have a similar issue. I'll try out a few stuff and keep you informed, but the basics are:

**Comment by AlexRamallo** [comment]
I am encountering this issue doing coincidentally a very similar thing to @lainwir3d, except instead of a btrfs snapshot I'm using restic to create an offsite backup of a nextcloud instance and its database. I did change my graphroot from the default to a raid1 btrfs with two drives, but that's about it.

The script is run by a systemd timer and it's basically just:

```
podman pod stop nextcloud
podman run ... restic/restic ...
podman pod start nextcloud
```

The backup container mounts the same volumes as the netxcloud containers (using :ro) in order to back them up. This happens both with/without SELinux, podman version is 4.6.2.

The only container that gets stuck is the nextcloud app, not the mariadb or redis containers.

If anyone wants to try reproducing it, this is just a standard nextcloud/mariadb/redis docker install using the latest tag for all three containers, and running on top of opensuse microos.

**Comment by lainwir3d** [comment]
Just after my last message, I did a `podman system migrate` and issue has not reproduced yet (ie. it's been fine for 4 days). I'll keep you informed if this good strike goes on.

**Comment by grooverdan** [comment]
I have a unkillable container (even after reboot)

```
[buildbot@bb-rhel8-docker ~]$ podman version
Client:       Podman Engine
Version:      4.4.1
API Version:  4.4.1
Go Version:   go1.19.12
Built:        Tue Sep 19 12:31:11 2023
OS/Arch:      linux/amd64
```

```
[buildbot@bb-rhel8-docker ~]$ podman info
host:
  arch: amd64
  buildahVersion: 1.29.0
  cgroupControllers: []
  cgroupManager: cgroupfs
  cgroupVersion: v1
  conmon:
    package: conmon-2.1.6-1.module+el8.8.0+19993+47c8ef84.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.6, commit: 0dcf8a206aff64a95e52d205642f8a7c7ed45049'
  cpuUtilization:
    idlePercent: 98.27
    systemPercent: 0.22
    userPercent: 1.52
  cpus: 4
  distribution:
    distribution: '"rhel"'
    version: "8.8"
  eventLogger: file
  hostname: bb-rhel8-docker
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1003
      size: 1
    - container_id: 1
      host_id: 296608
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1003
      size: 1
    - container_id: 1
      host_id: 296608
      size: 65536
  kernel: 4.18.0-477.21.1.el8_8.x86_64
  linkmode: dynamic
  logDriver: k8s-file
  memFree: 1616064512
  memTotal: 2113773568
  networkBackend: cni
  ociRuntime:
    name: runc
    package: runc-1.1.4-1.module+el8.8.0+19993+47c8ef84.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.1.4
      spec: 1.0.2-dev
      go: go1.19.12
      libseccomp: 2.5.2
  os: linux
  remoteSocket:
    path: /tmp/podman-run-1003/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_SYS_CHROOT,CAP_NET_RAW,CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: false
  serviceIsRemote: false
  slirp4netns:
    executable: /bin/slirp4netns
    package: slirp4netns-1.2.0-2.module+el8.8.0+19993+47c8ef84.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.4.0
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.2
  swapFree: 3221221376
  swapTotal: 3221221376
  uptime: 0h 15m 6.00s
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.access.redhat.com
  - registry.redhat.io
  - docker.io
store:
  configFile: /home/buildbot/.config/containers/storage.conf
  containerStore:
    number: 1
    paused: 0
    running: 0
    stopped: 1
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /home/buildbot/.local/share/containers/storage
  graphRootAllocated: 157934215168
  graphRootUsed: 1723305984
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "true"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 4
  runRoot: /tmp/containers-user-1003/containers
  transientStore: false
  volumePath: /home/buildbot/.local/share/containers/storage/volumes
version:
  APIVersion: 4.4.1
  Built: 1695119471
  BuiltTime: Tue Sep 19 12:31:11 2023
  GitCommit: ""
  GoVersion: go1.19.12
  Os: linux
  OsArch: linux/amd64
  Version: 4.4.1
```

```
[buildbot@bb-rhel8-docker ~]$ podman --log-level=debug ps -a
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called ps.PersistentPreRunE(podman --log-level=debug ps -a) 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /data/buildbot/containers/storage/libpod/bolt_state.db 
DEBU[0000] Overriding graph root "/data/buildbot/containers/storage" with "/home/buildbot/.local/share/containers/storage" from database 
DEBU[0000] Overriding static dir "/data/buildbot/containers/storage/libpod" with "/home/buildbot/.local/share/containers/storage/libpod" from database 
DEBU[0000] Overriding volume path "/data/buildbot/containers/storage/volumes" with "/home/buildbot/.local/share/containers/storage/volumes" from database 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/buildbot/.local/share/containers/storage 
DEBU[0000] Using run root /tmp/containers-user-1003/containers 
DEBU[0000] Using static dir /home/buildbot/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /tmp/podman-run-1003/libpod/tmp 
DEBU[0000] Using volume path /home/buildbot/.local/share/containers/storage/volumes 
DEBU[0000] Using transient store: false                 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] Not configuring container store              
DEBU[0000] Initializing event backend file              
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Configured OCI runtime ocijail initialization failed: no valid executable found for OCI runtime ocijail: invalid argument 
DEBU[0000] Configured OCI runtime crun initialization failed: no valid executable found for OCI runtime crun: invalid argument 
DEBU[0000] Configured OCI runtime runj initialization failed: no valid executable found for OCI runtime runj: invalid argument 
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Configured OCI runtime youki initialization failed: no valid executable found for OCI runtime youki: invalid argument 
DEBU[0000] Configured OCI runtime crun-wasm initialization failed: no valid executable found for OCI runtime crun-wasm: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/runc"            
INFO[0000] Setting parallel job count to 13             
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called ps.PersistentPreRunE(podman --log-level=debug ps -a) 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /data/buildbot/containers/storage/libpod/bolt_state.db 
DEBU[0000] Overriding run root "/tmp/podman-run-1003/containers" with "/tmp/containers-user-1003/containers" from database 
DEBU[0000] Overriding graph root "/data/buildbot/containers/storage" with "/home/buildbot/.local/share/containers/storage" from database 
DEBU[0000] Overriding static dir "/data/buildbot/containers/storage/libpod" with "/home/buildbot/.local/share/containers/storage/libpod" from database 
DEBU[0000] Overriding volume path "/data/buildbot/containers/storage/volumes" with "/home/buildbot/.local/share/containers/storage/volumes" from database 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/buildbot/.local/share/containers/storage 
DEBU[0000] Using run root /tmp/containers-user-1003/containers 
DEBU[0000] Using static dir /home/buildbot/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /tmp/podman-run-1003/libpod/tmp 
DEBU[0000] Using volume path /home/buildbot/.local/share/containers/storage/volumes 
DEBU[0000] Using transient store: false                 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that metacopy is not being used 
DEBU[0000] Cached value indicated that native-diff is usable 
DEBU[0000] backingFs=extfs, projectQuotaSupported=false, useNativeDiff=true, usingMetacopy=false 
DEBU[0000] Initializing event backend file              
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Configured OCI runtime ocijail initialization failed: no valid executable found for OCI runtime ocijail: invalid argument 
DEBU[0000] Configured OCI runtime crun initialization failed: no valid executable found for OCI runtime crun: invalid argument 
DEBU[0000] Configured OCI runtime crun-wasm initialization failed: no valid executable found for OCI runtime crun-wasm: invalid argument 
DEBU[0000] Configured OCI runtime runj initialization failed: no valid executable found for OCI runtime runj: invalid argument 
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime youki initialization failed: no valid executable found for OCI runtime youki: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/runc"            
INFO[0000] Setting parallel job count to 13             
DEBU[0000] Failed to add podman to systemd sandbox cgroup: exec: "dbus-launch": executable file not found in $PATH 
CONTAINER ID  IMAGE                                   COMMAND     CREATED       STATUS      PORTS                    NAMES
ba91b3031e89  localhost/mariadb-39581-aarch64:latest  mysqld      35 hours ago  Stopping    0.0.0.0:44257->3306/tcp  mariadbcontainer17254
DEBU[0000] Called ps.PersistentPostRunE(podman --log-level=debug ps -a) 
DEBU[0000] Shutting down engines                        
[buildbot@bb-rhel8-docker ~]$ podman --log-level=debug ps -a
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called ps.PersistentPreRunE(podman --log-level=debug ps -a) 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /data/buildbot/containers/storage/libpod/bolt_state.db 
DEBU[0000] Overriding graph root "/data/buildbot/containers/storage" with "/home/buildbot/.local/share/containers/storage" from database 
DEBU[0000] Overriding static dir "/data/buildbot/containers/storage/libpod" with "/home/buildbot/.local/share/containers/storage/libpod" from database 
DEBU[0000] Overriding volume path "/data/buildbot/containers/storage/volumes" with "/home/buildbot/.local/share/containers/storage/volumes" from database 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/buildbot/.local/share/containers/storage 
DEBU[0000] Using run root /tmp/containers-user-1003/containers 
DEBU[0000] Using static dir /home/buildbot/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /tmp/podman-run-1003/libpod/tmp 
DEBU[0000] Using volume path /home/buildbot/.local/share/containers/storage/volumes 
DEBU[0000] Using transient store: false                 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] Not configuring container store              
DEBU[0000] Initializing event backend file              
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime ocijail initialization failed: no valid executable found for OCI runtime ocijail: invalid argument 
DEBU[0000] Configured OCI runtime crun initialization failed: no valid executable found for OCI runtime crun: invalid argument 
DEBU[0000] Configured OCI runtime crun-wasm initialization failed: no valid executable found for OCI runtime crun-wasm: invalid argument 
DEBU[0000] Configured OCI runtime runj initialization failed: no valid executable found for OCI runtime runj: invalid argument 
DEBU[0000] Configured OCI runtime youki initialization failed: no valid executable found for OCI runtime youki: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/runc"            
INFO[0000] Setting parallel job count to 13             
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called ps.PersistentPreRunE(podman --log-level=debug ps -a) 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /data/buildbot/containers/storage/libpod/bolt_state.db 
DEBU[0000] Overriding run root "/tmp/podman-run-1003/containers" with "/tmp/containers-user-1003/containers" from database 
DEBU[0000] Overriding graph root "/data/buildbot/containers/storage" with "/home/buildbot/.local/share/containers/storage" from database 
DEBU[0000] Overriding static dir "/data/buildbot/containers/storage/libpod" with "/home/buildbot/.local/share/containers/storage/libpod" from database 
DEBU[0000] Overriding volume path "/data/buildbot/containers/storage/volumes" with "/home/buildbot/.local/share/containers/storage/volumes" from database 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/buildbot/.local/share/containers/storage 
DEBU[0000] Using run root /tmp/containers-user-1003/containers 
DEBU[0000] Using static dir /home/buildbot/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /tmp/podman-run-1003/libpod/tmp 
DEBU[0000] Using volume path /home/buildbot/.local/share/containers/storage/volumes 
DEBU[0000] Using transient store: false                 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that metacopy is not being used 
DEBU[0000] Cached value indicated that native-diff is usable 
DEBU[0000] backingFs=extfs, projectQuotaSupported=false, useNativeDiff=true, usingMetacopy=false 
DEBU[0000] Initializing event backend file              
DEBU[0000] Configured OCI runtime runj initialization failed: no valid executable found for OCI runtime runj: invalid argument 
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Configured OCI runtime ocijail initialization failed: no valid executable found for OCI runtime ocijail: invalid argument 
DEBU[0000] Configured OCI runtime youki initialization failed: no valid executable found for OCI runtime youki: invalid argument 
DEBU[0000] Configured OCI runtime crun initialization failed: no valid executable found for OCI runtime crun: invalid argument 
DEBU[0000] Configured OCI runtime crun-wasm initialization failed: no valid executable found for OCI runtime crun-wasm: invalid argument 
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/runc"            
INFO[0000] Setting parallel job count to 13             
DEBU[0000] Failed to add podman to systemd sandbox cgroup: exec: "dbus-launch": executable file not found in $PATH 
CONTAINER ID  IMAGE                                   COMMAND     CREATED       STATUS      PORTS                    NAMES
ba91b3031e89  localhost/mariadb-39581-aarch64:latest  mysqld      35 hours ago  Stopping    0.0.0.0:44257->3306/tcp  mariadbcontainer17254
DEBU[0000] Called ps.PersistentPostRunE(podman --log-level=debug ps -a) 
DEBU[0000] Shutting down engines                        



[buildbot@bb-rhel8-docker ~]$ podman kill --log-level=debug  -s 9 ba91b3031e89
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called kill.PersistentPreRunE(podman kill --log-level=debug -s 9 ba91b3031e89) 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /data/buildbot/containers/storage/libpod/bolt_state.db 
DEBU[0000] Overriding graph root "/data/buildbot/containers/storage" with "/home/buildbot/.local/share/containers/storage" from database 
DEBU[0000] Overriding static dir "/data/buildbot/containers/storage/libpod" with "/home/buildbot/.local/share/containers/storage/libpod" from database 
DEBU[0000] Overriding volume path "/data/buildbot/containers/storage/volumes" with "/home/buildbot/.local/share/containers/storage/volumes" from database 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/buildbot/.local/share/containers/storage 
DEBU[0000] Using run root /tmp/containers-user-1003/containers 
DEBU[0000] Using static dir /home/buildbot/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /tmp/podman-run-1003/libpod/tmp 
DEBU[0000] Using volume path /home/buildbot/.local/share/containers/storage/volumes 
DEBU[0000] Using transient store: false                 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] Not configuring container store              
DEBU[0000] Initializing event backend file              
DEBU[0000] Configured OCI runtime crun initialization failed: no valid executable found for OCI runtime crun: invalid argument 
DEBU[0000] Configured OCI runtime crun-wasm initialization failed: no valid executable found for OCI runtime crun-wasm: invalid argument 
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Configured OCI runtime youki initialization failed: no valid executable found for OCI runtime youki: invalid argument 
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Configured OCI runtime runj initialization failed: no valid executable found for OCI runtime runj: invalid argument 
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime ocijail initialization failed: no valid executable found for OCI runtime ocijail: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/runc"            
INFO[0000] Setting parallel job count to 13             
INFO[0000] podman filtering at log level debug          
DEBU[0000] Called kill.PersistentPreRunE(podman kill --log-level=debug -s 9 ba91b3031e89) 
DEBU[0000] Using conmon: "/usr/bin/conmon"              
DEBU[0000] Initializing boltdb state at /data/buildbot/containers/storage/libpod/bolt_state.db 
DEBU[0000] Overriding run root "/tmp/podman-run-1003/containers" with "/tmp/containers-user-1003/containers" from database 
DEBU[0000] Overriding graph root "/data/buildbot/containers/storage" with "/home/buildbot/.local/share/containers/storage" from database 
DEBU[0000] Overriding static dir "/data/buildbot/containers/storage/libpod" with "/home/buildbot/.local/share/containers/storage/libpod" from database 
DEBU[0000] Overriding volume path "/data/buildbot/containers/storage/volumes" with "/home/buildbot/.local/share/containers/storage/volumes" from database 
DEBU[0000] Using graph driver overlay                   
DEBU[0000] Using graph root /home/buildbot/.local/share/containers/storage 
DEBU[0000] Using run root /tmp/containers-user-1003/containers 
DEBU[0000] Using static dir /home/buildbot/.local/share/containers/storage/libpod 
DEBU[0000] Using tmp dir /tmp/podman-run-1003/libpod/tmp 
DEBU[0000] Using volume path /home/buildbot/.local/share/containers/storage/volumes 
DEBU[0000] Using transient store: false                 
DEBU[0000] Set libpod namespace to ""                   
DEBU[0000] [graphdriver] trying provided driver "overlay" 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that overlay is supported 
DEBU[0000] Cached value indicated that metacopy is not being used 
DEBU[0000] Cached value indicated that native-diff is usable 
DEBU[0000] backingFs=extfs, projectQuotaSupported=false, useNativeDiff=true, usingMetacopy=false 
DEBU[0000] Initializing event backend file              
DEBU[0000] Configured OCI runtime krun initialization failed: no valid executable found for OCI runtime krun: invalid argument 
DEBU[0000] Configured OCI runtime crun initialization failed: no valid executable found for OCI runtime crun: invalid argument 
DEBU[0000] Configured OCI runtime runj initialization failed: no valid executable found for OCI runtime runj: invalid argument 
DEBU[0000] Configured OCI runtime kata initialization failed: no valid executable found for OCI runtime kata: invalid argument 
DEBU[0000] Configured OCI runtime runsc initialization failed: no valid executable found for OCI runtime runsc: invalid argument 
DEBU[0000] Configured OCI runtime youki initialization failed: no valid executable found for OCI runtime youki: invalid argument 
DEBU[0000] Configured OCI runtime crun-wasm initialization failed: no valid executable found for OCI runtime crun-wasm: invalid argument 
DEBU[0000] Configured OCI runtime ocijail initialization failed: no valid executable found for OCI runtime ocijail: invalid argument 
DEBU[0000] Using OCI runtime "/usr/bin/runc"            
INFO[0000] Setting parallel job count to 13             
DEBU[0000] Failed to add podman to systemd sandbox cgroup: dial unix /run/user/0/bus: connect: permission denied 
DEBU[0000] Sending signal 9 to container ba91b3031e89e93ff81b97a259ecc5accd58ec9d26e73adbcca63ff7a96773b6 
ERRO[0000] EOF                                          
INFO[0000] Error updating status for container ba91b3031e89e93ff81b97a259ecc5accd58ec9d26e73adbcca63ff7a96773b6: decoding container status for container ba91b3031e89e93ff81b97a259ecc5accd58ec9d26e73adbcca63ff7a96773b6: EOF 
Error: sending signal to container ba91b3031e89e93ff81b97a259ecc5accd58ec9d26e73adbcca63ff7a96773b6: `/usr/bin/runc kill ba91b3031e89e93ff81b97a259ecc5accd58ec9d26e73adbcca63ff7a96773b6 9` failed: exit status 1
DEBU[0000] Shutting down engines                        
```


```
[buildbot@bb-rhel8-docker ~]$ podman inspect mariadbcontainer17254
Error: getting graph driver info "ba91b3031e89e93ff81b97a259ecc5accd58ec9d26e73adbcca63ff7a96773b6": readlink /home/buildbot/.local/share/containers/storage/overlay: invalid argument
```

**Comment by grooverdan** [comment]
Might be relevant - deleting dangling links:
```
[dan@bb-rhel8-docker ~]$ sudo -u root find -L /home/buildbot/.local/share/containers/storage/overlay  -type l -delete
...
find: File system loop detected; ‘/home/buildbot/.local/share/containers/storage/overlay/l/URFCA4BNOF6U7LN5DJZEDACNBU/usr/share/doc/node-path-parse/node-resolve/node-types-resolve/node-resolve’ is part of the same file system loop as ‘/home/buildbot/.local/share/containers/storage/overlay/l/URFCA4BNOF6U7LN5DJZEDACNBU/usr/share/doc/node-path-parse/node-resolve’.
find: File system loop detected; ‘/home/buildbot/.local/share/containers/storage/overlay/l/URFCA4BNOF6U7LN5DJZEDACNBU/usr/share/doc/node-path-parse/node-resolve/node-path-parse’ is part of the same file system loop as ‘/home/buildbot/.local/share/containers/storage/overlay/l/URFCA4BNOF6U7LN5DJZEDACNBU/usr/share/doc/node-path-parse’.
```

**Comment by hickersonj** [comment]
I'm having the same issue, if I perform a `kill -9` on `conmon`, my private registry container is stuck in the `stopping` state even when I restart the podman service. 

```
podman ps -a
CONTAINER ID  IMAGE                                              COMMAND               CREATED       STATUS      PORTS                    NAMES
a308d3387664  docker.io/library/registry:latest                  /etc/docker/regis...  3 months ago  Stopping    0.0.0.0:5000->5000/tcp   registry
```

Rebooting my Linux machine (Fedora 33) did resolve it, so that's at least less egregious than deleting the private registry and re-deploying it. But from others comments, a reboot doesn't always resolve the issue.

```
podman version
Client:       Podman Engine
Version:      4.6.0
API Version:  4.6.0
Go Version:   go1.19.6
Git Commit:   38e6fab9664c6e59b66e73523b307a56130316ae
Built:        Wed Jul 26 18:19:45 2023
OS/Arch:      linux/amd64
```

**Comment by grooverdan** [comment]
> Might be relevant - deleting dangling links:
> 
> ```
> [dan@bb-rhel8-docker ~]$ sudo -u root find -L /home/buildbot/.local/share/containers/storage/overlay  -type l -delete
> ...
> find: File system loop detected; ‘/home/buildbot/.local/share/containers/storage/overlay/l/URFCA4BNOF6U7LN5DJZEDACNBU/usr/share/doc/node-path-parse/node-resolve/node-types-resolve/node-resolve’ is part of the same file system loop as ‘/home/buildbot/.local/share/containers/storage/overlay/l/URFCA4BNOF6U7LN5DJZEDACNBU/usr/share/doc/node-path-parse/node-resolve’.
> find: File system loop detected; ‘/home/buildbot/.local/share/containers/storage/overlay/l/URFCA4BNOF6U7LN5DJZEDACNBU/usr/share/doc/node-path-parse/node-resolve/node-path-parse’ is part of the same file system loop as ‘/home/buildbot/.local/share/containers/storage/overlay/l/URFCA4BNOF6U7LN5DJZEDACNBU/usr/share/doc/node-path-parse’.
> ```

Actually this ^ was a big mistake.

I ended up doing a buildah clean up and then podman cleanup and that seemed to clear it.

**Comment by vrothberg** [comment]
@mheon, I find this one tricky.  It seems to happen on multiple sites but without a clear reproducer AFAICS.

**Comment by mcepl** [comment]
I have a container, where something which looks awfully like this happens 100%. It is created with this distrobox definition:

```
[vis]
image="registry.opensuse.org/home/mcepl/moldavite/containers/opensuse/vis-master:latest"
pull=true
replace=true
```

If use it as my `$EDITOR` through this script:

```
#!/bin/sh
# distrobox_binary
# name: vis
if [ -z "${CONTAINER_ID}" ]; then
        exec /usr/bin/distrobox-enter  -n vis --                 /usr/bin/vis  "$@"
elif [ -x "/usr/bin/vis" ] ; then
    /usr/bin/vis "$@"
elif [ -x "/usr/bin/distrobox-host-exec" ] ; then
    /usr/bin/distrobox-host-exec ~/.bin/vis "$@"
else
    echo "Cannot run vis"
    exit 255
fi
```

Whenever I want to upgrade the container (`distrobox assemble create -n vis`) it gets stuck. See also https://github.com/89luca89/distrobox/issues/1301

**Comment by mheon** [comment]
Also tracked in #19629 closing in favor of that one.
Fix incoming in 5.1 regardless.
